R/ResamplingHoldout.R

```diff
@@ -59,10 +61,11 @@ ResamplingHoldout = R6Class("ResamplingHoldout", inherit = Resampling,
     }
   ),
   private = list(
-    .sample = function(ids, ...) {
+    .sample = function(ids, task, weights, ...) {
+      pv = self$param_set$values
       n = length(ids)
       in_train = logical(n)
-      in_train[sample.int(n, round(n * self$param_set$values$ratio))] = TRUE
+      in_train[sample.int(n, round(n * self$param_set$values$ratio), prob = weights)] = TRUE
```

Comment:
Is this here really the semantics that we want?
A task with 10 obs, ratio = 0.8 and a probability vector `c(0, 0, 0, 0, 0, 1, 1, 1, 1, 1)` will not produce a valid resampling split. I am not sure how intuitive this weight semantic is for resamplings that don't allow for duplicate observations in the training data.